### PR TITLE
add cloud func code

### PR DESCRIPTION
**Pull request by wleofeng**
@kevinlin505

### CODE REVIEW

File: functions/firebase-debug.log [hunk 1]
```diff
@@ -0,0 +1,27 @@
+[debug] [2019-07-11T23:33:16.266Z] ----------------------------------------------------------------------
```
Comment (kevinlin505) [inline functions/firebase-debug.log]: gitignore this file

File: functions/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
+{
+  "parserOptions": {
```
Comment (kevinlin505) [inline functions/.eslintrc.json]: you can copy and paste the one from main project.

File: .firebaserc [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "projects": {
-    "default": "goplan-3b4b1"
+    "default": "goplan-3b4b1",
+    "dev": "goplan-3b4b1"
```
Comment (kevinlin505) [inline .firebaserc]: where do we run the function deploy command?  inside the root directory of the main project or inside `function/`?

Comment (wleofeng) [reply .firebaserc]: inside /function , i will add instructions

Comment (wleofeng) [reply .firebaserc]: added in Readme

**Pull request merged**